registry/remote/repository.go

```
@@ -392,20 +402,24 @@ func (r *Repository) Referrers(ctx context.Context, desc ocispec.Descriptor, art
 	}
 
 	err := r.referrersByAPI(ctx, desc, artifactType, fn)
-	if state == referrersStateSupported {
-		// The repository is known to support Referrers API, no fallback.
-		return err
-	}
-
-	// The referrers state is unknown.
 	if err != nil {
-		if errors.Is(err, errdef.ErrNotFound) {
-			// A 404 returned by Referrers API indicates that Referrers API is
-			// not supported. Fallback to referrers tag schema.
-			r.SetReferrersCapability(false)
-			return r.referrersByTagSchema(ctx, desc, artifactType, fn)
+		if !errcode.IsErrorResponseStatus(err, http.StatusNotFound) {
+			return err
 		}
-		return err
+
+		if errcode.IsErrorCode(err, errcode.ErrorCodeNameUnknown) {
+			// The repository name is not found, no fallback
+			return fmt.Errorf("%s: %w", r.Reference.Repository, errdef.ErrNotFound)
```

Should we return error in this case of just hide the error for users?